src/read.h

```diff
@@ -27,18 +27,30 @@
 
 #ifndef DEBUG_READ_ERROR
 
-#define READ_ERROR()    (TLS(NrError) || (TLS(NrError)+=sySetjmp(TLS(ReadJmpError))))
+#define TRY_READ \
+    if (!TLS(NrError)) { \
+        if (sySetjmp(TLS(ReadJmpError))) { \
+            TLS(NrError)++; \
+        }\
+    }\
+    if (!TLS(NrError))
 
 #else
 
-#define READ_ERROR()                                                     \
-    ( TLS(NrError) ||                                                         \
-      ( ( TLS(NrError) += setjmp(TLS(ReadJmpError)) ) ?                            \
-        Pr( "READ_ERROR( %s, %d )\n", (Int)__FILE__, __LINE__ ),0 : 0 ), \
-      TLS(NrError) )
+#define TRY_READ \
+    if (!TLS(NrError)) { \
+        if (sySetjmp(TLS(ReadJmpError))) { \
+            Pr( "READ_ERROR( %s, %d )\n", (Int)__FILE__, __LINE__ ); \
+            TLS(NrError)++; \
+        }\
+    }\
+    if (!TLS(NrError)) {
 
 #endif
 
+#define CATCH_READ_ERROR \
+    else
+
```

Comment:
Is it worth calling this `CATCH_READ_ERROR` instead of just leaving it as `else`?